gix-index/src/entry/stat.rs

```diff
@@ -0,0 +1,415 @@
+use std::cmp::Ordering;
+use std::time::{SystemTime, SystemTimeError};
+
+use filetime::FileTime;
+
+use crate::entry::Stat;
+
+impl Stat {
+    /// detect whether this stat entry is racy given the timestamp
+    /// of the index it belongs to
```

Comment:
I expanded the documentation in 4736b6061acd28abfe8872a10f803f42e0591e35. Hope that's clearer now.

Reply:
Yes, thanks a lot! In general, if the modification date of an entry is before the index's modification date, it can be trusted, but if it's at the same time as the index was written, it could have changed after it was seen by the index. And if that's the case, additional checking needs to be done and there is some special handling around zero-sized entries, but that I will hopefully grasp later once I am with `gix-worktree` :).